ui/app/templates/components/keymgmt/distribute.hbs

```diff
@@ -151,24 +151,16 @@
       </div>
     </fieldset>
 
-    <div class="field is-grouped box is-fullwidth is-bottomless">
-      <div class="control">
-        <button
-          type="submit"
-          disabled={{or this.validationErrorCount this.createDistribution.isRunning}}
-          class="button is-primary"
-          data-test-secret-save={{true}}
-        >
-          {{#if this.createDistribution.isRunning}}
-            <span class="loader is-inline-block"></span>
-          {{else}}
-            {{if (or (not @key) this.isNewKey) "Add key" "Distribute key"}}
-          {{/if}}
-        </button>
-      </div>
-      <div class="control">
-        <Hds::Button @text="Cancel" @color="secondary" {{on "click" @onClose}} />
-      </div>
-    </div>
+    <hr class="has-background-gray-100" />
+    <Hds::ButtonSet>
+      <Hds::Button
+        @text={{if (or (not @key) this.isNewKey) "Add key" "Distribute key"}}
+        @icon={{if this.saveModel.isRunning "loading"}}
+        type="submit"
+        data-test-secret-save={{true}}
+        disabled={{or this.validationErrorCount this.createDistribution.isRunning}}
```

Comment:
We no longer disable buttons when there are validation errors so this first condition can be removed
```suggestion
        disabled={{this.createDistribution.isRunning}}
```